### PR TITLE
Transport timer fix

### PR DESCRIPTION
**Pull request by Aenge**
Increases reliability of ships and elevators through the use of state machines. Also removes handling of the Port Bastok drawbridge through elevators. Instead, it uses a periodic trigger. This trigger system was intended for NPC's that could benefit from time-based scripting other than OnGameHour. In addition to the drawbridge, other NPC's, such as those related to transportation that shout things periodically, could be handled with it. I'm not sure if there was something like this already in place, so let me know if it was a bad idea. 

Also, I'm new to git, be gentle ~

### CODE REVIEW

**Comment by teschnei** [comment]
Thanks for the merge request. Currently I'm locked out of my account
because my 2fa is old, so it might be a few days before I can review it.
Just as a small nitpick to fix until then, I would like to see all
`operator new`s replaced with either unique_ptr (in the case of the
singleton) or just stack variables (in the case of trigger vector)

File: scripts/globals/status.lua [hunk 1]
```diff
@@ -2462,7 +2462,7 @@ dsp.elevator =
     TIMED_AUTOMATIC           = 0,
     DAVOI_LIFT                = 1,
     PALBOROUGH_MINES_LIFT     = 2,
-    FORT_GHELSBA_LIFT         = 3
+    FORT_GHELSBA_LIFT         = 3,
```
[the remark this answers is not in the document]
Comment (Aenge) [reply scripts/globals/status.lua]: This was added for consistency, as most of the tables in status.lua use a comma after the last entry. I can remove it if you prefer

[the remark this answers is not in the document]
Comment (TeoTwawki) [reply scripts/globals/status.lua]: convention is to only put it between elements, it has no effect either way when its the last one

File: src/map/lua/lua_baseentity.cpp [hunk 2]
```diff
@@ -2532,7 +2575,54 @@ inline int32 CLuaBaseEntity::isInMogHouse(lua_State* L)
...
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    DSP_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_NPC);
+
+    DSP_DEBUG_BREAK_IF(lua_isnil(L, 1) || !lua_isnumber(L, 1));
+    DSP_DEBUG_BREAK_IF(lua_isnil(L, 2) || !lua_isnumber(L, 2));
+
+    CNpcEntity* PNpc = (CNpcEntity*)m_PBaseEntity;
+
+    PNpc->status = (STATUSTYPE)lua_tointeger(L, 1);
+    PNpc->animation = (UINT8)lua_tointeger(L,2);
```
Comment (takhlaq) [inline src/map/lua/lua_baseentity.cpp]: uint8

**Comment by Aenge** [comment]
I tried to rebase my branch, looks like I messed up this PR?

**Comment by TeoTwawki** [comment]
messed up your branch. if you fix the branch the pr will auto update to whatever is in it when you do a force push.

**Comment by Aenge** [comment]
Thanks @TeoTwawki

File: scripts/zones/Port_Bastok/npcs/_6kt.lua [hunk 3]
```diff
@@ -6,16 +6,52 @@ require("scripts/globals/status");
...
+function onTimeTrigger(npc, triggerID)
+--npc is the moghouse-side door to the bridge
+    local upperDoor = GetNPCByID(npc:getID() - 1)
+    local bridge = GetNPCByID(npc:getID() - 2)
 
-    npc:setElevator(elevator.id, elevator.lowerDoor, elevator.upperDoor, elevator.elevator, elevator.started, elevator.regime);
-end;
\ No newline at end of file
+    if npc == nil or upperDoor == nil or bridge == nil then
+        error("Drawbridge NPC failure")
```
Comment (TeoTwawki) [inline scripts/zones/Port_Bastok/npcs/_6kt.lua]: :thinking:

**Pull request merged**